### PR TITLE
feat: formblock api

### PR DESCRIPTION
**Pull request by csiyang**
# Description

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 507105f</samp>

This pull request adds a new feature to the api-journeys and api-gateway apps that allows users to create and update blocks of type `FormBlock`, which embed forms from Formium in their journeys. It also updates the schema, the resolvers, and the tests for the existing action types to support `FormBlock`. It modifies the `Block` model and table to store the Formium credentials, and adds a new scalar type `Json` to handle the form data. It also improves the documentation and formatting of the `adminJourneys` query field.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/35391169/todos/6813111447)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 507105f</samp>

*  Add the FormBlock type and its resolvers to allow users to embed a form from Formium in their journeys ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-e60078ed0fc71b8cd3bb2fca25505e7dba866b7d463147662a12432dfe287273R397-R424), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-e60078ed0fc71b8cd3bb2fca25505e7dba866b7d463147662a12432dfe287273R958-R960), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-e60078ed0fc71b8cd3bb2fca25505e7dba866b7d463147662a12432dfe287273R1103-R1110), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-e60078ed0fc71b8cd3bb2fca25505e7dba866b7d463147662a12432dfe287273L1343-R1389), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-18e5f83e2a677613317ecf9a5565f08285958770d93903e178f1ecc6a5fc8fa7R1-R4), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-881378863f13d5e9573c2c2e756ae61e8161918ebb4a384a8d5dddd7bf7385b0R433-R435), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-3cbd0d77b8eff95fb0f62db7f9f1a4b69ccd568ae3902a3450bac7d752c817a3R15-R16), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-3cbd0d77b8eff95fb0f62db7f9f1a4b69ccd568ae3902a3450bac7d752c817a3R147-R154), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-3cbd0d77b8eff95fb0f62db7f9f1a4b69ccd568ae3902a3450bac7d752c817a3R454-R474), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-3cbd0d77b8eff95fb0f62db7f9f1a4b69ccd568ae3902a3450bac7d752c817a3L1748-R1786), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R296-R307), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R821-R832), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R1407-R1410), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R1572), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-2c297b432319911516c210c9e2c03d42fafcd7567c02282d9da0fe21d877ac92R1), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-d33f2cff54146883293444aaeb54bffac5bb7f586a79bf7eb0ae19f573d47626R1-R24), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-77cccc59977d3bb4dee6ceed05b31eb063f1c1c9c54bd86af258fb85724ec409R1-R19), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-af885295911bd27bdfa4a1824d68a4497fdfd815b070f8ba988ac9dece3089c0R12), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-af885295911bd27bdfa4a1824d68a4497fdfd815b070f8ba988ac9dece3089c0R31), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-d5b96a7739d28f79a295c0ab80121047cb7d99774d51731b4e5ce2cdf9eedbbcR1-R32), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-258c02420ba5fd9d78807064f72cbcdadc6e10ae5ed52d2470fb1869b4ab1ad3R1-R204), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-1b75bd440b78c3a12b2b00c613080827b4ddf0f6387a78d7a3adf133ad1d50c9R1-R110))
  * Add the type definition for FormBlock to the `schema.graphql` file in the `api-gateway` app, implementing the Block interface and adding join directives for the JOURNEYS graph ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-e60078ed0fc71b8cd3bb2fca25505e7dba866b7d463147662a12432dfe287273R397-R424))
  * Add the scalar type Json to the `schema.graphql` file in the `api-gateway` app, adding a join directive for the JOURNEYS graph ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-e60078ed0fc71b8cd3bb2fca25505e7dba866b7d463147662a12432dfe287273R958-R960))
  * Add the mutation fields for formBlockCreate and formBlockUpdate to the `schema.graphql` file in the `api-gateway` app, taking input types for the required and optional fields for FormBlocks and adding join directives for the JOURNEYS graph ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-e60078ed0fc71b8cd3bb2fca25505e7dba866b7d463147662a12432dfe287273R1103-R1110))
  * Modify the adminJourneys query field to add a description for the useLastActiveTeamId argument and format the field for readability in the `schema.graphql` file in the `api-gateway` app ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-e60078ed0fc71b8cd3bb2fca25505e7dba866b7d463147662a12432dfe287273L1343-R1389))
  * Add a SQL migration script to alter the Block table and add columns for the Formium credentials: projectId, apiToken, and formSlug in the `migration.sql` file in the `api-journeys` app ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-18e5f83e2a677613317ecf9a5565f08285958770d93903e178f1ecc6a5fc8fa7R1-R4))
  * Add the Formium credentials fields to the Block model in the `schema.prisma` file in the `api-journeys` app ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-881378863f13d5e9573c2c2e756ae61e8161918ebb4a384a8d5dddd7bf7385b0R433-R435))
  * Add the scalar type Json to the `schema.graphql` file in the `api-journeys` app ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-3cbd0d77b8eff95fb0f62db7f9f1a4b69ccd568ae3902a3450bac7d752c817a3R15-R16))
  * Add the mutation fields for formBlockCreate and formBlockUpdate to the `schema.graphql` file in the `api-journeys` app, without the join directives ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-3cbd0d77b8eff95fb0f62db7f9f1a4b69ccd568ae3902a3450bac7d752c817a3R147-R154))
  * Add the type definition for FormBlock to the `schema.graphql` file in the `api-journeys` app, without the join directives ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-3cbd0d77b8eff95fb0f62db7f9f1a4b69ccd568ae3902a3450bac7d752c817a3R454-R474))
  * Modify the adminJourneys query field to add a description for the useLastActiveTeamId argument and format the field for readability in the `schema.graphql` file in the `api-journeys` app, without the join directives ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-3cbd0d77b8eff95fb0f62db7f9f1a4b69ccd568ae3902a3450bac7d752c817a3L1748-R1786))
  * Add the classes for FormBlockCreateInput and FormBlockUpdateInput to the `graphql.ts` file in the `api-journeys` app, generated from the input types in the schema ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R296-R307))
  * Add the class for FormBlock to the `graphql.ts` file in the `api-journeys` app, generated from the type definition in the schema ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R821-R832))
  * Add the abstract methods for formBlockCreate and formBlockUpdate to the IMutation class in the `graphql.ts` file in the `api-journeys` app, generated from the mutation fields in the schema ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R1407-R1410))
  * Add the type alias for Json to the `graphql.ts` file in the `api-journeys` app, generated from the scalar type in the schema ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-bef781d45ccbb6775960a589c940a77584bd143e501a8ae9b83d699d803561f2R1572))
  * Add the `json.graphql` file to the `api-journeys` app, containing the scalar type definition for Json ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-2c297b432319911516c210c9e2c03d42fafcd7567c02282d9da0fe21d877ac92R1))
  * Add the `json.provider.ts` file to the `api-journeys` app, containing the JsonScalar class that implements the CustomScalar interface and provides the methods for serializing and parsing the Json scalar type ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-d33f2cff54146883293444aaeb54bffac5bb7f586a79bf7eb0ae19f573d47626R1-R24))
  * Add the `json.spec.ts` file to the `api-journeys` app, containing the unit tests for the JsonScalar class, using Jest and NestJS Testing ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-77cccc59977d3bb4dee6ceed05b31eb063f1c1c9c54bd86af258fb85724ec409R1-R19))
  * Add the import statement for the FormBlockResolver class to the `block.module.ts` file in the `api-journeys` app ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-af885295911bd27bdfa4a1824d68a4497fdfd815b070f8ba988ac9dece3089c0R12))
  * Add the FormBlockResolver class to the providers array in the BlockModule in the `block.module.ts` file in the `api-journeys` app ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-af885295911bd27bdfa4a1824d68a4497fdfd815b070f8ba988ac9dece3089c0R31))
  * Add the `form.graphql` file to the `api-journeys` app, containing the type definition and the mutation fields for the FormBlock type ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-d5b96a7739d28f79a295c0ab80121047cb7d99774d51731b4e5ce2cdf9eedbbcR1-R32))
  * Add the `form.resolver.spec.ts` file to the `api-journeys` app, containing the unit tests for the FormBlockResolver class, using Jest, NestJS Testing, and mock implementations of the PrismaService, the BlockService, and the FormiumClient ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-258c02420ba5fd9d78807064f72cbcdadc6e10ae5ed52d2470fb1869b4ab1ad3R1-R204))
  * Add the `form.resolver.ts` file to the `api-journeys` app, containing the FormBlockResolver class that implements the resolvers for the FormBlock type and its fields, using the PrismaService, the BlockService, and the FormiumClient as dependencies, and the CaslAbility and the AppCaslGuard to check the authorization of the user ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-1b75bd440b78c3a12b2b00c613080827b4ddf0f6387a78d7a3adf133ad1d50c9R1-R110))
* Include FormBlock in the list of block types that can have various actions, such as email, link, navigate, navigate to block, and navigate to journey, in the action resolvers in the `api-journeys` app ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-63e5dfbf51551571ecf00f17b3ba0141a316f2662a8748f44565cb9c872771ccL66-R67), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-4bd62b4b4f9247353282dc562aaffb4c252d2a29310a259230ffcbff80b6b6b9L63-R64), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-f3b132bb9916362d5366a0c2ed49f7ecf2937ebaf86fd5f980e6a063e3167976L58-R59), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-ab465086e377597f3b1ab1f22ac1438c336135dc974f831a74c157a33e4d6f51L58-R59), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-7b370de65d7fc9c3fe9888c076c3979e62c0f63b7b6499eeafeb9eeba7802b74L59-R60), [link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-667b420f6bc05fedc3288c096d34930c07bb96f2f60729f63926e116f511314aL59-R60))
  * Modify the ActionResolver class to include FormBlock in the list of block types that can have an action in the `action.resolver.ts` file ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-63e5dfbf51551571ecf00f17b3ba0141a316f2662a8748f44565cb9c872771ccL66-R67))
  * Modify the EmailActionResolver class to include FormBlock in the list of block types that can have an email action in the `emailAction.resolver.ts` file ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-4bd62b4b4f9247353282dc562aaffb4c252d2a29310a259230ffcbff80b6b6b9L63-R64))
  * Modify the LinkActionResolver class to include FormBlock in the list of block types that can have a link action in the `linkAction.resolver.ts` file ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-f3b132bb9916362d5366a0c2ed49f7ecf2937ebaf86fd5f980e6a063e3167976L58-R59))
  * Modify the NavigateActionResolver class to include FormBlock in the list of block types that can have a navigate action in the `navigateAction.resolver.ts` file ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-ab465086e377597f3b1ab1f22ac1438c336135dc974f831a74c157a33e4d6f51L58-R59))
  * Modify the NavigateToBlockActionResolver class to include FormBlock in the list of block types that can have a navigate to block action in the `navigateToBlockAction.resolver.ts` file ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-7b370de65d7fc9c3fe9888c076c3979e62c0f63b7b6499eeafeb9eeba7802b74L59-R60))
  * Modify the NavigateToJourneyActionResolver class to include FormBlock in the list of block types that can have a navigate to journey action in the `navigateToJourneyAction.resolver.ts` file ([link](https://github.com/JesusFilm/core/pull/2105/files?diff=unified&w=0#diff-667b420f6bc05fedc3288c096d34930c07bb96f2f60729f63926e116f511314aL59-R60))

### CODE REVIEW

File: apps/api-journeys/src/app/modules/block/block.resolver.ts [hunk 1]
```diff
@@ -112,4 +112,27 @@ export class BlockResolver {
       })
     return await this.blockService.removeBlockAndChildren(block)
   }
+
+  @Query()
+  async block(@Args('id') id: string): Promise<Block> {
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/block.resolver.ts:117]: must check if user has permission to administrate the block

File: apps/api-journeys/src/app/modules/block/form/form.graphql [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+}
+
 type FormBlock implements Block {
   id: ID!
   journeyId: ID!
   parentBlockId: ID
   parentOrder: Int
   action: Action
   form: Json
+  projects: [Project!]
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.graphql:13]:
```suggestion
  projects: [FormiumProject!]
```

File: apps/api-journeys/src/app/modules/block/form/form.graphql [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
   projectId: String
+  forms: [String!]
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.graphql:15]:
```suggestion
  forms: [FormiumForm!]
```

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 4]
```diff
@@ -93,10 +94,51 @@ export class FormBlockResolver {
     return await this.blockService.update(id, input)
   }
 
+  @ResolveField('projects')
+  async projects(@Parent() block: Block): Promise<Project[] | null> {
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:98]:
```suggestion
  async projects(@Parent() block: Block): Promise<Project[]> {
```

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 5]
```diff
@@ -93,10 +94,51 @@ export class FormBlockResolver {
...
+        id: project.id,
+        name: project.name
+      }))
+    } catch (e) {
+      return null
+    }
+  }
+
+  @ResolveField('forms')
+  async forms(@Parent() block: Block): Promise<string[] | null> {
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:117]:
```suggestion
  async forms(@Parent() block: Block): Promise<string[]> {
```

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/api-journeys/src/app/modules/block/form/form.graphql (hunk 3 above)
Comment (csiyang) [reply apps/api-journeys/src/app/modules/block/form/form.graphql:15]: Forms is only meant to be an array of strings right? Are we wanting to store more than just the slugs here?

File: apps/api-journeys/src/app/modules/block/form/form.graphql [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  The name of the project
+  """
+  name: String!
+}
+
+type FormiumForm {
+  """
+  The formSlug of the form
+  """
+  id: String!
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.graphql:16]:
```suggestion
  slug: String!
```

File: apps/api-journeys/src/app/modules/block/form/form.graphql [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+type FormBlock implements Block {
+  id: ID!
+  journeyId: ID!
+  parentBlockId: ID
+  parentOrder: Int
+  action: Action
+  form: Json
+  projects: [FormiumProject!]
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.graphql:30]:
```suggestion
  projects: [FormiumProject!]!
```

File: apps/api-journeys/src/app/modules/block/form/form.graphql [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  projectId: String
+  forms: [FormiumForm!]
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.graphql:32]:
```suggestion
  forms: [FormiumForm!]!
```

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+    })
+    if (block == null)
+      throw new GraphQLError('block not found', {
+        extensions: { code: 'NOT_FOUND' }
+      })
+    if (!ability.can(Action.Update, subject('Journey', block.journey)))
+      throw new GraphQLError('user is not allowed to update block', {
+        extensions: { code: 'FORBIDDEN' }
+      })
+    return await this.blockService.update(id, input)
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:95]: still no validation of inputs here.

It should verify that apiToken, projectId, and/or formSlug are valid before saving.

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+  @ResolveField('projects')
+  async projects(@Parent() block: Block): Promise<FormiumProject[]> {
+    const { apiToken } = block
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:100]: you can move this into the line above.

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+    if (apiToken === null) return []
+
+    try {
+      const projectsData =
+        (await (await createClient('', { apiToken })).getMyProjects()).data ??
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:105]: createClient doesn't need to be awaited

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+        []
+
+      return projectsData.map((project) => ({
+        id: project.id,
+        name: project.name
+      }))
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:111]:
```suggestion
      return projectsData.map(({ id, name }) => ({ id, name }))
```

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+  @ResolveField('forms')
+  async forms(@Parent() block: Block): Promise<FormiumForm[]> {
+    const { projectId, apiToken } = block
+    if (projectId == null || apiToken == null) return []
+
+    try {
+      const formsData =
+        (
+          await (
+            await createClient(projectId, {
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:126]:
```suggestion
            createClient(projectId, {
```

does not need to be awaited

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+              apiToken
+            })
+          ).findForms()
+        ).data ?? []
+
+      return formsData.map((form) => ({
+        id: form.slug,
+        name: form.name
+      }))
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:135]:
```suggestion
      return formsData.map(({ slug, name }) => ({ slug, name }))
```

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+        name: project.name
+      }))
+    } catch (e) {
+      return []
+    }
+  }
+
+  @ResolveField('forms')
+  async forms(@Parent() block: Block): Promise<FormiumForm[]> {
+    const { projectId, apiToken } = block
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:119]: can be moved to line above

File: apps/api-journeys/src/app/modules/block/form/form.resolver.ts [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+        name: form.name
+      }))
+    } catch (e) {
+      return []
+    }
+  }
+
+  @ResolveField('form')
+  async form(@Parent() block: Block): Promise<Json | null> {
+    const { projectId, apiToken, formSlug } = block
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/form.resolver.ts:143]: can be destructured in line above

**Review by tataihono — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 17]
```diff
@@ -73,7 +73,7 @@
     }
   ],
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
```
Comment (tataihono) [inline .vscode/settings.json:76]: revert

File: apps/api-journeys/src/app/modules/block/form/validateFormCredentials/validateFormCredentials.ts [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+
+export async function validateFormCredentials({
+  input,
+  block
+}: ValidateFormCredentialsProps): Promise<FormBlockUpdateInput> {
+  const combinedApiToken = input.apiToken ?? block.apiToken
+  const combinedProjectId = input.projectId ?? block.projectId
+  const inputApiToken = 'apiToken' in input
+  const inputProjectId = 'projectId' in input
+  const inputFormSlug = 'formSlug' in input
```
Comment (tataihono) [inline apps/api-journeys/src/app/modules/block/form/validateFormCredentials/validateFormCredentials.ts:21]:
```suggestion
  const inputApiTokenPresent = 'apiToken' in input
  const inputProjectIdPresent = 'projectId' in input
  const inputFormSlugPresent = 'formSlug' in input
```

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**